File: pset1/mario/less/mario.c
#include<stdio.h>
#include<cs50.h>

int main(void)
{
    int h;
    //first submit a integer, then judge if it is authorized
    do
    {
        printf("Height:");
        h = get_int();
    }
    while (h < 0 || h > 23);


    if (h == 0) {}
    else
    {
        //authorized number of stairs
        //print blocks by #-symbol
        for (int i = 0; i < h; i++)//every layer has h+1 blocks
        {
            for (int space = 0; space < h - 1 - i; space++)
            {
                printf(" ");
            }
            for (int block = 0; block < i + 2; block++)
            {
                printf("#");
            }
            printf("\n");
        }
    }
}

File: pset6/mario.py
import cs50
import sys
def main():
    print("Height:")
    h = cs50.get_int()
    while(h < 0 or h > 23):
        print("Height:")
        h = cs50.get_int()
    print("{}".format(h))

    if h != 0:
        for i in range(h):
            for space in range(h-1-i):
                print("", end="")
            for block in range(i+2):
                print("#", end="")
            print("");
    exit(1);

if __name__ == "__main__":
    main()